### DISCUSSION TITLE
Add a `toRecipient` method to allow sending mail notifications to a different email without overriding routeNotificationForMail method

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#46661] [Ideas] [answered]
I have a Laravel 10 project which allows a user to customise the email address that should receive alerts from my system. This, right now is done by overriding the `routeNotificationForMail` method on my `User` model. However, when the user types in a different email address that isn't their account's email address, important emails such as password resets and other emails are sent to this notification email which may not be their account's email address and thus means users cannot reset their passwords.

I only really need to override the email that should receive alerts in my `toMail` method of some notifications, but I don't see anywhere where this is supported by Laravel? Here's one of my `toMail` method in a `DomainExpiry` notification class:

```php
/**
 * Get the mail representation of the notification.
 *
 * @param  mixed  $notifiable
 */
public function toMail($notifiable)
{
    $domain = isset($this->emailData[0]['domain']) ? $this->emailData[0]['domain'] : '';
    $domain = htmlspecialchars($domain);

    $data = [
        'is_subscribed' => $notifiable->is_subscribed,
        'first_name' => $notifiable['first_name'],
        'domains' => json_encode($this->emailData),
    ];

    try {
        $this->createHistoryEntry('mail', 'Domain Expiry', $data, $notifiable);
    } catch (\Exception $e) {
    }

    return (new MailMessage)
                ->subject(isset($domain) && ! empty($domain) ? "Domain Expiry Alert: $domain" : 'Domain Expiry Alert')
                ->markdown('emails.domains.expiry', $data);
}
```

And here's the method overrideen on the `User` model:

```php
/**
 * Route notifications for the mail channel.
 *
 * @return array|string
 */
public function routeNotificationForMail(Notification $notification)
{
    return $this->notification_email;
}
```

How do I customise the [recipient](https://laravel.com/docs/10.x/notifications#customizing-the-recipient) without sending important alerts to this `notification_email` field?

Is there a way to fetch the user model into the `toMail` and set a recipient right on:

```php
return (new MailMessage)
            ->subject('My subject')
            ->markdown('emails.domains.expiry', $data);
```

Maybe there's a `toRecipient` method?

**Comment by Angel5a** [comment, marked as answer]
You can return a full Mailable. See [Using Mailables](https://laravel.com/docs/10.x/notifications#using-mailables) section.

**Reply by sts-ryan-holton** [reply]
@Angel5a cool! And this would allow me to use markdown etc?

**Reply by Angel5a** [reply]
Yes it is.
